include/uk/essentials.h

```diff
@@ -360,6 +360,10 @@ extern "C" {
 #define UK_NARGS(...) \
 	__UK_NARGS_X(, ##__VA_ARGS__, 25, 24, 23, 22, 21, 20, 19, 18, 17, 16, \
 		     15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0)
+#define UK_NARGS_HALF(...) \
```

Comment:
`UK_NARGS` is generic enough to keep it here. It is just a way for using the pre-processor to count the number of arguments that a variadic macro got handed over. This information can be used to build more complex macros on top. `UK_NARGS_HALF` looks to me syscall_shim specific because I remember we had a macro like this originally in syscall_shim but I was able to rewrite the macros in syscall_shim to base them also on `UK_NARGS`. Is it really needed for your case?

Reply:
I use it for `UK_USR_CALLMAPx`. Thought I would pass the actual number of arguments.. but I guess I could just use the double of that and just pretend I am using the actual number of args but I will have to have a look again because I don't remember much.